File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/scala/redocpot/spark/Reduce.scala
package scala.redocpot.spark

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object Reduce {
  def main(args : Array[String]) : Unit = {
    val conf = new SparkConf().setAppName("Reduce").setMaster("local[*]")
    val sc = new SparkContext(conf)
    val input = sc.parallelize(List(1, 2, 3, 4))
    val sum = input.reduce((x, y) => x + y)
    println(sum)
    sc.stop
  }
}

File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/java/redocpot/spark/Reduce.java
package redocpot.spark;

import java.util.Arrays;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function2;

public class Reduce {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
            .setMaster("local")
            .setAppName("Reduce");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<Integer> rdd = sc.parallelize(Arrays.asList(1, 2, 3, 4));
        Integer sum = rdd.reduce(new Function2<Integer, Integer, Integer>() {
            public Integer call(Integer x, Integer y) {
                return x + y;
            }
        });
        System.out.println(sum);

        sc.stop();
    }
}
